### PR TITLE
feat(tabs): add underline tab variation

### PR DESCRIPTION
**Pull request by daanishnasir**
BREAKING CHANGE:

Default of tabs is now underline, pass in tabType=‘pills’ for pill variation, module cleanup

## Description

- Removed empty spec files, moved tests and added more tests to a single test file
- Cleaned up modules, only TabsModule is needed now. 
- Add underline tabs as the default variation 
- Add tabType prop for pill variation 
- focus fix and onKeyHandling works

## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: angular/src/lib/tabs/tab-header.component.ts [hunk 1]
```diff
@@ -1,22 +1,18 @@
...
 
-import { Component, OnInit, Input } from '@angular/core';
+import { Component, Input } from '@angular/core';
 
 @Component({
   selector: 'md-tab-header',
   template: `
     {{ heading }}
-    <div *ngIf="subHeading">{{ subHeading }}</div>
+    <div *ngIf="subHeading" class="md-tab-header--subheading">{{ subHeading }}</div>
```
Comment (bfbiggs) [inline angular/src/lib/tabs/tab-header.component.ts]: Would this make more sense to either be md-tab-header__subheading as it's an element?

File: angular/src/lib/tabs/tabs.component.ts [hunk 2]
```diff
@@ -21,21 +19,22 @@ export type TabType = 'pills';
...
+  /** @prop Type of Tabs | 'pills' */
+  @Input() tabType: string =  '';
+  /** @prop Callback function invoked when user selects a tab | null */
   @Output() whenSelect = new EventEmitter();
 
   @HostBinding('class') get classes(): string {
     return (
       'md-tab' +
       `${(this.tabType && ` md-tab--${this.tabType}`) || ''}` +
+      `${(!this.tabType && ' md-tab--tabs md-tab--nav') || ''}` +
```
Comment (bfbiggs) [inline angular/src/lib/tabs/tabs.component.ts]: So should this maybe just be the default or underline tabType?  This creates a breaking change for intending to use pills. @pauljeter

Comment (pauljeter) [reply angular/src/lib/tabs/tabs.component.ts]: It makes more sense for the Tabs default to be this style vs the pill buttons. I am ok with the breaking change

File: angular/src/lib/tabs/tab-header.component.ts (hunk 1 above)
Comment (pauljeter) [reply angular/src/lib/tabs/tab-header.component.ts]: I do agree that the className here should be `md-tab-header__subheading`

**Comment by bfbiggs** [comment]
Ideally we would knock out react but let's atleast update the HTML examples.

File: angular/src/lib/tabs/tabs.service.ts [hunk 3]
```diff
@@ -84,7 +86,7 @@ export class TabsService {
   }
 
   public setFocusLast = () => {
-    this.setFocus(this.Tabs - 1);
+    this.setFocus(this.Tabs);
```
Comment (ArthusLiang) [inline angular/src/lib/tabs/tabs.service.ts]: Why change this default value to -1 ? Is that for supporting none showing tab?

**Pull request merged**